File: helm/relayforge/src/relayforge/test_sink/state.py
"""Глобальное состояние test-sink (одно на Pod).

Отдельный модуль — чтобы избежать circular import между test_sink.app
и test_sink.routes. Счётчики попыток и факт применения хранятся в
ReceiptStore (файлы на emptyDir) и переживают restart контейнера.
"""


class SinkState:
    def __init__(self):
        self.verification_key: bytes = b""
        self.control_token: bytes = b""
        self.modes = None  # test_sink.modes.ModeController
        self.receipts = None  # test_sink.receipts.ReceiptStore


state = SinkState()

File: helm_golang/relayforge/internal/testsink/state.go
package testsink

// SinkState is the global state of the test-sink, populated during Run and used
// by the HTTP handlers (mirrors test_sink/state.py).
type SinkState struct {
	VerificationKey []byte
	ControlToken    []byte
	Modes           *ModeController
	Receipts        *ReceiptStore
}

// State is the shared instance used by routes handlers.
var State = &SinkState{}
